packages/wrangler/src/containers/config.ts

```diff
@@ -46,6 +46,81 @@ function convertContainerAffinitiesForApi(
  * We set defaults here too, because we can assume that if the value is undefined,
  * we want to revert to the default rather than inheriting from the prev deployment
  */
+
+
+/**
+ * Auto-generate Durable Object bindings and migrations for containers
+ * This reduces wrangler.toml boilerplate by creating necessary DO config automatically
+ */
+export const autoGenerateContainerDurableObjects = (config: Config): void => {
+	if (!config.containers || config.containers.length === 0) {
+		return;
+	}
+
+	// Initialize durable_objects structure if it doesn't exist
+	if (!config.durable_objects) {
+		config.durable_objects = { bindings: [] };
+	}
+	if (!config.durable_objects.bindings) {
+		config.durable_objects.bindings = [];
+	}
+
+	// Initialize migrations if it doesn't exist
+	if (!config.migrations) {
+		config.migrations = [];
+	}
+
+	for (const container of config.containers) {
+		// Auto-generate class_name if not provided
+		if (!container.class_name) {
+			// Convert container name to PascalCase for class name
+			container.class_name = container.name
+				.split(/[-_]/)
+				.map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+				.join('') + 'Container';
+		}
+
+		// Auto-generate DO binding if it doesn't exist
+		const existingBinding = config.durable_objects.bindings.find(
+			binding => binding.class_name === container.class_name
+		);
+		
+		if (!existingBinding) {
+			// Generate binding name from container name (SCREAMING_SNAKE_CASE)
+			const bindingName = container.name.toUpperCase().replace(/[-]/g, '_');
+			
+			config.durable_objects.bindings.push({
+				class_name: container.class_name,
+				name: bindingName,
+			});
+		}
+
+		// Auto-generate migration if class doesn't exist in migrations
+		const hasExistingMigration = config.migrations.some(migration =>
+			migration.new_sqlite_classes?.includes(container.class_name!)
+		);
+
+		if (!hasExistingMigration) {
+			// Find existing v1 migration or create new one
+			let v1Migration = config.migrations.find(migration => migration.tag === 'v1');
```

Comment:
IIUC `v1` is another convention and users might have an existing valid migration with a different label.

If that's correct, please document the convention your code is using